pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -65,19 +63,20 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req kube_ctrl.Reque
 		return kube_ctrl.Result{}, errors.Wrap(err, "could not upsert GatewayRoute")
 	}
 
-	var err error
-
-	resource.Spec, err = r.gapiToKumaRoute(ctx, mesh, httpRoute.Namespace, httpRoute)
-	if err != nil {
-		return kube_ctrl.Result{}, err
+	if err := r.updateStatus(ctx, httpRoute); err != nil {
```

Comment:
I'm not sure about this approach. We are trying to statelessly convert the HTTPRoute to a GatewayRoute and then check everything after the fact. Maybe it would be better to not try to split the responsibilities like this. For example, the converter can't set status when it finds an unsupported reference kind.

WDYT?

Reply:
Yep, I was depending on creating a harmless invalid GatewayRoute but tracking the state while converting is a better pattern.

Reply:
Done